cmd/neofs-cli/modules/object/util.go

```diff
@@ -358,6 +383,154 @@ func finalizeSession(cmd *cobra.Command, dst SessionPrm, tok *session.Object, ke
 	return nil
 }
 
+// CreateSessionV2 creates V2 token locally and
+// writes the result into the provided SessionPrm.
+//
+// SessionPrm MUST be one of:
+//
+//	*internal.PutObjectPrm
+//	*internal.DeleteObjectPrm
+func CreateSessionV2(ctx context.Context, cmd *cobra.Command, dst SessionPrm, cli *client.Client, key *ecdsa.PrivateKey, cnr cid.ID, objs ...oid.ID) error {
+	common.PrintVerbose(cmd, "Creating V2 session token locally...")
+	nm, err := cli.NetMapSnapshot(ctx, client.PrmNetMapSnapshot{})
+	if err != nil {
+		return fmt.Errorf("can't get NetMap snapshot: %w", err)
+	}
+
+	common.PrintVerbose(cmd, "Getting container info...")
+	cnrObj, err := cli.ContainerGet(ctx, cnr, client.PrmContainerGet{})
+	if err != nil {
+		return fmt.Errorf("can't get container: %w", err)
+	}
+
+	policy := cnrObj.PlacementPolicy()
+
+	common.PrintVerbose(cmd, "Getting container nodes from NetMap...")
+	nodes, err := nm.ContainerNodes(policy, cnr)
+	if err != nil {
+		return fmt.Errorf("can't get container nodes: %w", err)
+	}
+
+	var subjects []sessionv2.Target
+	nodeCount := 0
+
+	for _, replica := range nodes {
+		for _, node := range replica {
+			neoPubKey, err := keys.NewPublicKeyFromBytes(node.PublicKey(), elliptic.P256())
+			if err != nil {
+				common.PrintVerbose(cmd, "Warning: failed to parse node public key: %v", err)
+				continue
+			}
+
+			ecdsaPubKey := (*ecdsa.PublicKey)(neoPubKey)
+			userID := user.NewFromECDSAPublicKey(*ecdsaPubKey)
+
+			subjects = append(subjects, sessionv2.NewTargetUser(userID))
+			nodeCount++
+		}
+	}
+
+	common.PrintVerbose(cmd, "Added %d node public keys as subjects", nodeCount)
```

Comment:
I'd expect v2 to accept targets as parameters. Can be a single node, can be a set of nodes, can be NNS, can be whatever.